### DISCUSSION TITLE
Repr-type of rustified translated C enum?

### DISCUSSION THREAD

**Discussion by boozook** [rust-lang/rust-bindgen#2806] [Q&A] [answered]
Hello 👋

How to determine, preferably before bindings generation, translated rust type of enum?

On various archs and OSs it can be __unsigned__ or __signed__ integer (`i32`/`u32`/etc) and I need to determine its `repr` somehow.

The size also depending on args like "short-enum", but I need some default.

**Comment by pvdrz** [comment, marked as answer]
> How to determine, preferably before bindings generation, translated rust type of enum?

I think we don't provide any way to do this, maybe we could extend `ParseCallbacks` to do such thing.

> On various archs and OSs it can be **unsigned** or **signed** integer (`i32`/`u32`/etc) and I need to determine its `repr` somehow.
> 
> The size also depending on args like "short-enum", but I need some default.

In either case `bindgen` will use the size that `clang` decides, so if you pass `-- -fshort-enums`  to bindgen, it will behave accordingly.
